dashboards-notebooks/public/components/paragraph_components/para_query_grid.tsx

```diff
@@ -136,4 +136,13 @@ export function QueryDataGrid(props: QueryDataGridProps) {
       />
     </div>
   )
-}
\ No newline at end of file
+}
+
+function queryDataGridPropsAreEqual(prevProps: QueryDataGridProps, nextProps: QueryDataGridProps) {
+  return prevProps.rowCount === nextProps.rowCount
+    && JSON.stringify(prevProps.queryColumns) === JSON.stringify(nextProps.queryColumns)
+    && JSON.stringify(prevProps.visibleColumns) === JSON.stringify(nextProps.visibleColumns)
+    && JSON.stringify(prevProps.dataValues) === JSON.stringify(nextProps.dataValues)
+}
+
+export const QueryDataGridMemo = React.memo(QueryDataGrid, queryDataGridPropsAreEqual);
```

Comment:
since the memorized component is exported here, maybe remove the original export on line 47? So it won't be accidentally used

Reply:
Good catch, will remove